ui/text_selection.js

```diff
@@ -164,10 +164,10 @@ shaka.ui.TextSelection = class extends shaka.ui.SettingsMenu {
    * @private
    */
   async onTextTrackSelected_(track) {
-    await this.player.setTextTrackVisibility(true);
     if (this.player) {  // May have become null while awaiting
       this.player.selectTextLanguage(track.language, track.roles[0]);
     }
+    await this.player.setTextTrackVisibility(true);
```

Comment:
Moving the visibility change after selectTextLanguage() removes the need to check for `this.player`.  We know it's non-null at the start of the method, but it may become null in the middle of an async call if someone calls `destroy()`.

But I'm also not entirely sure why you changed this.  Can you explain?

Reply:
Oh, sorry, now I see your comment from before:

> I found another bug (which I now fixed) while fixing this issue, where captions show up in 2 different languages simultaneously when turned off, then back on in a different language. This happens because the UI library calls setTextTrackVisibility then selectTextLanguage, which both try to switch the text stream to different languages. However, if we reverse the order in which they are called, then when setTextTrackVisibility is called after selectTextLanguage, it will see that a language is already selected, and won't try to switch the text stream, fixing the issue!

Then please remove the `if (this.player)` above and leave a comment explaining why these calls need to be in this order.  Otherwise, we may change it back some day without realizing we broke anything.  (Even better would be an automated test for this, but I would accept a strongly-worded comment in this case.)